src/data/writeToStore.ts

```
@@ -195,6 +197,31 @@ export function writeSelectionSetToStore({
   return store;
 }
 
+// Checks if the id given is an id that was generated by Apollo
+// rather than by dataIdFromObject.
+function isGeneratedId(id: string): boolean {
+  return (id[0] === '$');
+}
+
+function mergeWithGenerated(generatedKey: string, realKey: string, cache: NormalizedCache) {
+  const generated = cache[generatedKey];
+  const real = cache[realKey];
+
+  if (!isObject(generated) || !isObject(real)) {
+    return;
```

This condition could happen if you have a particular key like "name" that has a value that looks like "$David" or something similar, i.e. this error could happen if the data returned by the server uses a `$` sign in a particular way. I don't think the client should impose restrictions on the data returned so I guess it makes sense to silently ignore that issue.